classes.py

```diff
@@ -0,0 +1,544 @@
+        if self.this_games < 5:
+            return False
+        elif self.this_games == 5:
+            return self.this_points_above_expected >= 1.84
+        elif self.this_games == 6:
+            return self.this_points_above_expected >= 2.02
+        elif self.this_games == 7:
+            return self.this_points_above_expected >= 2.16
+        raise NotImplementedError(
+            'Unknown condition for Rating performance rule with more than 7 games.')
+
+    def check_double_k_rule(self):
+        if self.this_games < 4:
+            return False
+        elif self.this_games == 4:
+            return self.this_points_above_expected >= 1.65
+        elif self.this_games == 5:
+            return self.this_points_above_expected >= 1.43
+        elif self.this_games == 6:
+            return self.this_points_above_expected >= 1.56
+        elif self.this_games == 7:
+            return self.this_points_above_expected >= 1.69
+        raise NotImplementedError(
+            'Unknown condition for double K rule with more than 7 games.')
+
+    def get_current_k(self):
+        # Assumes rating is not temporary
+        for (k, starting_num_games) in _K_STARTING_NUM_GAMES:
+            if self.last_total_games >= starting_num_games:
+                current_k = k
+        return current_k
+
+    def get_performance_rating(self, avg_oppon_rating, num_valid_games, total_num_points):
+        # In case of perfect results, consider score as if there was an extra game that ended in a
+        # draw.
+        score = total_num_points / num_valid_games
+        if score == 1.0:
+            score = (num_valid_games + 0.5) / (num_valid_games + 1.0)
+        elif score == 0.0:
+            score = 0.5 / (num_valid_games + 1.0)
+        return avg_oppon_rating + 400.0 * math.log10(score / (1.0 - score))
+
+
+class Tournament:
+
+    def __init__(self, rating_cycle, tournament):
+        self.ord = int(tournament[0])
+        self.id = int(tournament[1])
+        self.name = tournament[2]
+        self.date_end = tournament[3]
+        self.type = tournament[4]
+        self.is_irt = int(tournament[5])
+        self.is_fexerj = int(tournament[6])
+        self.players = {}
+        self.unrated_keys = []
+        self.temp_keys = []
+        self.established_keys = []
+        self.rating_cycle = rating_cycle
+        self.load_player_list()
+
+    def load_player_list(self):
+        pass
+
+    def complete_players_info(self):
+        for snr, tp in self.players.items():
+            if self.is_irt:
+                fp = self.rating_cycle.rating_list[self.rating_cycle.cbx_to_fexerj[tp.id]]
+            else:
+                fp = self.rating_cycle.rating_list[tp.id]
+
+            tp.last_rating = int(fp.lastRating)
+            tp.last_total_games = int(fp.totGames)
+            tp.last_sum_oppon_ratings = int(fp.sumOppRating)
+            tp.last_pts_against_oppon = float(fp.ptsAgainstOpp)
+
+            if int(fp.totGames) == 0:
```

Comment:
fp.totGames should either always be an int everywhere, or this cast should be done only once and stored in a local variable